File: xhd/src/main/java/com/architecture/light/ui/dialog/LoadingDialog.java
package com.architecture.light.ui.dialog;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.StringRes;

import com.android.architecture.ui.page.BaseDialog;
import com.architecture.light.R;

/**
 * File describe:
 * Author: SuQi
 * Create date: 2022/8/13
 * Modify date: 2022/8/13
 * Version: 1
 */
public final class LoadingDialog {

    public static final class Builder extends BaseDialog.Builder<Builder> {

        private final TextView mMessageView;

        public Builder(Context context) {
            super(context);
            setContentView(R.layout.dialog_loading);
            setAnimStyle(BaseDialog.ANIM_TOAST);
            setBackgroundDimEnabled(false);
            setCancelable(false);

            mMessageView = findViewById(R.id.tv_wait_message);
        }

        public Builder setMessage(@StringRes int id) {
            return setMessage(getString(id));
        }

        public Builder setMessage(CharSequence text) {
            mMessageView.setText(text);
            mMessageView.setVisibility(text == null ? View.GONE : View.VISIBLE);
            return this;
        }
    }
}

File: app/src/main/java/com/architecture/light/ui/dialog/LoadingDialog.kt
package com.architecture.light.ui.dialog

import android.content.Context
import android.view.View
import android.widget.TextView
import com.android.architecture.extension.empty
import com.android.architecture.ui.page.BaseDialog
import com.architecture.light.R

/**
 * File describe:
 * Author: SuQi
 * Create date: 2022/8/13
 * Modify date: 2022/8/13
 * Version: 1
 */
class LoadingDialog {

    class Builder(context: Context) : BaseDialog.Builder<Builder>(context) {

        private val tvMessage: TextView

        init {
            setContentView(R.layout.dialog_loading)
            setAnimStyle(BaseDialog.ANIM_TOAST)
            setBackgroundDimEnabled(false)
            setCancelable(false)
            tvMessage = findViewById(R.id.tv_wait_message)
        }

        fun message(text: String): Builder {
            tvMessage.text = text
            tvMessage.visibility = if (text.empty) View.GONE else View.VISIBLE
            return this
        }

    }
}